ethcore/src/miner/mod.rs

```diff
@@ -84,7 +84,7 @@ pub trait MinerService : Send + Sync {
 
 	/// Update current pending block
 	fn update_sealing<C>(&self, chain: &C)
-		where C: BlockChain + CallContract + BlockProducer + SealedBlockImporter + Nonce + Sync;
+		where C: BlockChainClient;
```

Comment:
I think originally this was the bound, but we have moved to more relaxed traits to avoid tight coupling with `Client`. I'd say it would be good to keep it this way.